frontend/src/app/Settings/DeviceSettings.jsx

```diff
@@ -18,10 +22,17 @@ const getAllDevices = gql`
 `;
 
 const DeviceSettings = ({ deviceSettings, updateDeviceSettings }) => {
+  const appInsights = useAppInsightsContext();
+  const trackFetchDeviceSettings = useTrackEvent(appInsights, "Fetch Devices");
+
+  useEffect(() => {
```

Comment:
Another way to do this would using a higher-order component that does the reporting to wrap these components.